hawkbit-ddi-resource/src/main/java/org/eclipse/hawkbit/ddi/rest/resource/DdiRootController.java

```diff
@@ -326,7 +344,10 @@ private ActionStatusCreate generateUpdateStatus(final DdiActionFeedback feedback
         }
 
         if (feedback.getStatus().getDetails() != null) {
-            messages.addAll(feedback.getStatus().getDetails());
+            messages.addAll(feedback.getStatus()
```

Comment:
I don't think that you can do it that way. First of all people may rely on the structure of these messages (as you intend to do). So they might rely on the fact that a DDI provided message "A" is represented in management API as "A". The second problem is that we haven't used the time field yet as we have an API problem there (see #191). Last but not least I don't think that we should store this field in every message in the DB, kind of inefficient.

Long story short if you need this time field in your use-case I kindly ask you to introduce a new DB column for ActionStatus, store the time in there and fix the time field in the API (see #191). I am not sure however how to properly return it in your DDI extension. I personally would go for:

1) First guaranteeing that the time field is filled in the DB (i.e. by setting it with server TIME if not provided by controller).
2) Structure the messages in your extension by the time (this way the controller even sees which messages belonged together).